app/models/task.py

```diff
@@ -1,5 +1,37 @@
 from app import db
 
-
 class Task(db.Model):
-    task_id = db.Column(db.Integer, primary_key=True)
+    id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    title = db.Column(db.String, nullable=False)
+    description = db.Column(db.String, nullable=False)
+    completed_at = db.Column(db.DateTime, nullable=True)
+    goal = db.relationship("Goal", back_populates="tasks")
+    goal_id = db.Column(db.Integer, db.ForeignKey("goal.id"))
+    
+    def to_dict_with_goal(self):
+        if not self.completed_at:
+            self.is_complete = False
+        return (dict(
+                id=self.id,
+                title=self.title,
+                description=self.description,
+                is_complete=self.is_complete,
+                goal_id=self.goal_id
+            ))
+    def to_dict(self):
```

Comment:
Rather than have two functions that do nearly the same thing, why not adopt the same approach of using conditional logic to set the `goal_id`?